packages/electron/src/electron-main/electron-application.ts

```diff
@@ -0,0 +1,388 @@
+/********************************************************************************
+ * Copyright (C) 2020 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { ContributionProvider } from '@theia/core/lib/common/contribution-provider';
+import { MaybePromise } from '@theia/core/lib/common/types';
+import URI from '@theia/core/lib/common/uri';
+import { ElectronSecurityToken } from '@theia/core/lib/electron-common/electron-token';
```

Comment:
The `WebviewWidgetFactory` is contributed by `plugin-ext`, I don't think it would make sense to move that code to `electron`. This would also make `plugin-ext` dependent on `electron` which is annoying. It looks to me it is best to leave things as they are for now :(